sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/BlobPartitionManager.cs

```diff
@@ -189,83 +172,106 @@ public override async Task<IEnumerable<PartitionOwnership>> ClaimOwnershipAsync(
+                    // TODO: Add log  - "Ownership with partition id = '{ ownership.PartitionId }' claimed."
                 }
                 catch (RequestFailedException ex) when (ex.ErrorCode == BlobErrorCode.ConditionNotMet)
                 {
-                    Log($"Ownership with partition id = '{ ownership.PartitionId }' is not claimable.");
+                    // TODO: Add log  - "Ownership with partition id = '{ ownership.PartitionId }' is not claimable."
                 }
             }
 
             return claimedOwnership;
         }
 
         /// <summary>
-        ///   Updates the checkpoint using the given information for the associated partition and consumer group in the storage blob service.
+        ///    List of all the checkpoints in a data store for a given namespace, eventhub and consumer group.
         /// </summary>
         ///
-        /// <param name="checkpoint">The checkpoint containing the information to be stored.</param>
+        /// <param name="fullyQualifiedNamespace">The fully qualified Event Hubs namespace the ownership are associated with.  This is likely to be similar to <c>{yournamespace}.servicebus.windows.net</c>.</param>
+        /// <param name="eventHubName">The name of the specific Event Hub the ownership are associated with, relative to the Event Hubs namespace that contains it.</param>
+        /// <param name="consumerGroup">The name of the consumer group the ownership are associated with.</param>
         ///
-        /// <returns>A task to be resolved on when the operation has completed.</returns>
+        /// <returns>An enumerable containing all the existing checkpoints for the associated Event Hub and consumer group.</returns>
         ///
-        public override async Task UpdateCheckpointAsync(Checkpoint checkpoint)
+        public override async Task<IEnumerable<Checkpoint>> ListCheckpointsAsync(string fullyQualifiedNamespace,
+                                                                                       string eventHubName,
+                                                                                       string consumerGroup)
         {
-            var blobName = $"{ checkpoint.FullyQualifiedNamespace }/{ checkpoint.EventHubName }/{ checkpoint.ConsumerGroup }/{ checkpoint.PartitionId }";
-            BlobClient blobClient = _containerClient.GetBlobClient(blobName);
-
-            BlobProperties currentBlob;
-
+            List<Checkpoint> checkpoints = new List<Checkpoint>();
             try
             {
-                currentBlob = (await blobClient.GetPropertiesAsync().ConfigureAwait(false)).Value;
+                var prefix = $"{ fullyQualifiedNamespace }/{ eventHubName }/{ consumerGroup }/checkpoint/";
+                await foreach (BlobItem blob in _containerClient.GetBlobsAsync(traits: BlobTraits.Metadata, prefix: prefix).ConfigureAwait(false))
+                {
+                    long offset = 0;
+                    long sequenceNumber = 0;
```

Comment:
I don't know if 0 is a good default value in case they are not found (I think one of them, not sure which one, can assume this value in a normal scenario). @jsquire Should we use a negative number here? Or make them nullable? Or is 0 good?